src/main/java/com/redhat/devtools/lsp4ij/client/features/LSPCompletionProposal.java

```diff
@@ -100,14 +114,16 @@ public void handleInsert(@NotNull InsertionContext context) {
         updateCompletionItemFromResolved();
         Template template = null;
         if (item.getInsertTextFormat() == InsertTextFormat.Snippet) {
+            // Adjust the snippet content as appropriate based on client config
+            String snippetContent = adjustSnippetContent(getInsertText());
```

Comment:
Now we adjust the LSP-provided snippet before parsing it into a template if appropriate based on client config and the snippet contents.